x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/type/DataTypeConversionTests.java

```diff
@@ -33,17 +44,28 @@
 import static org.elasticsearch.xpack.sql.type.DataType.values;
 import static org.elasticsearch.xpack.sql.type.DataTypeConversion.commonType;
 import static org.elasticsearch.xpack.sql.type.DataTypeConversion.conversionFor;
+import static org.elasticsearch.xpack.sql.util.DateUtils.asDateTime;
 
 
 public class DataTypeConversionTests extends ESTestCase {
-    public void testConversionToString() {
-        Conversion conversion = conversionFor(DOUBLE, KEYWORD);
-        assertNull(conversion.convert(null));
-        assertEquals("10.0", conversion.convert(10.0));
 
-        conversion = conversionFor(DATETIME, KEYWORD);
-        assertNull(conversion.convert(null));
-        assertEquals("1970-01-01T00:00:00.000Z", conversion.convert(dateTime(0)));
+    public void testConversionToString() {
+        DataType to = KEYWORD;
+        {
+            Conversion conversion = conversionFor(DOUBLE, to);
+            assertNull(conversion.convert(null));
+            assertEquals("10.0", conversion.convert(10.0));
+        }
+        {
+            Conversion conversion = conversionFor(DATE, to);
+            assertNull(conversion.convert(null));
+            assertEquals("1973-11-29", conversion.convert(DateUtils.asDateOnly(123456789101L)));
+        }
+        {
+            Conversion conversion = conversionFor(DATETIME, to);
+            assertNull(conversion.convert(null));
+            assertEquals("1973-11-29T21:33:09.101Z", conversion.convert(asDateTime(123456789101L)));
```

Comment:
Can you add a test for a negative long, as well? And not only in this test, but everywhere where numbers and dates are involved.